### ISSUE TITLE
evaluation commands of fine-grained instruction tuning models

### ISSUE THREAD

**Issue by Chy2023** [nnnth/UFO#40]
Thank you for this nice work.

Similar to #39 , I need a config/ckpt to support multiple tasks within one model, and `instruction tuning models` seem to be a good choice. However, issue #39 closed as completed without mentioning the solution.
The provided training and evaluation [script](https://github.com/nnnth/UFO/blob/main/docs/en/UFO_commands.md) does not include evaluation of  `instruction tuning models` (e.g. [ufo-internvl2-8b-instruction.pth](https://huggingface.co/kanashi6/UFO/blob/main/ufo-internvl2-8b-instruction.pth)), it only contains evaluation commands of `specific finetuning models`. So I wonder if you could explain how to evaluate `instruction tuning models` to support multiple tasks.

By the way, I have another question regarding ckpt files. The ckpt for `multi-task training` [ufo-internvl2-8b-multi-448](https://huggingface.co/kanashi6/UFO/tree/main/ufo-internvl2-8b-multi-448) have a heavy size of 32.3 GB, while its backbone [InternVL2.5-8B](https://huggingface.co/OpenGVLab/InternVL2_5-8B) is 16.2 GB and `instruction tuning` [ufo-internvl2-8b-instruction.pth](https://huggingface.co/kanashi6/UFO/blob/main/ufo-internvl2-8b-instruction.pth) is 3.25 GB.
How could this happen? Since `instruction tuning` use LoRA and only train LLM part, I guess the ckpt only contains the updated weights? But I don't see relavant codes about only saving `required_grad=True` params in [training config](https://github.com/nnnth/UFO/blob/main/configs/UFO-InternVL2_5-8B/internvl2_5_8b_instruction_12w.py), it seems that all params are saved. 
Why is the size of `multi-task training` model much larger than its backbone? Although all its params are trained, its [model config](https://github.com/nnnth/UFO/blob/main/mmdet/models/detectors/ufo_internvl_full.py) and [training config](https://github.com/nnnth/UFO/blob/main/configs/UFO-InternVL2_5-8B-multitask/multi_448_30w.py) seem similar to those of `instruction tuning` model.

**Comment by nnnth** [comment]
Sorry for the late reply: 
1. The multi-task model has only been trained on datasets such as coco and ade20k, and it is full-parameter training, so the generalization of the model may not be very good. For example, if you want to perform open vocabulary detection or general vqa tasks, the effect may not be very good. 
2.  The multi-task evaluation is actually fake training. The number of training steps in the config is only 1, and the learning rate is also very low. This is done to make the model trained for one step and the loaded model basically consistent. The reason for this is that the test memory overhead of the multi-task model is very large, so I need to use deepspeed when testing, but I don’t figure out how to use deepspeed for testing on the mmdetection framework. However, it is normal to use deepspeed for training and then test during training, so this somewhat strange approach was adopted. 
3.  For multi-task training models, the loading process is to first use the official InternVL2_5-8B, and then overwrite it with the model weights we provide. For the instruction model, since we use lora during training, the loading process is to first use the official InternVL2_5-8B, then add the lora weights, and then load the model weights we provide, which will only overwrite the lora weights. 
4. Unfortunately, my GPU resources are currently in short supply, so I can't help you confirm the memory overhead. For the demo not running, I think it may be because the test resolution is too high. We use a resolution of 896x896 by default. I think adjusting it to 448x448 may allow it to run in 3090. To adjust the resolution, you need to modify the `resize` operation in `refer_test_pipeline` in config.
